addenda18_test.go

```diff
@@ -170,121 +172,92 @@ func BenchmarkAddenda18String(b *testing.B) {
 func TestValidateAddenda18RecordType(t *testing.T) {
 	addenda18 := mockAddenda18()
 	addenda18.recordType = "63"
-	if err := addenda18.Validate(); err != nil {
-		if e, ok := err.(*FieldError); ok {
-			if e.FieldName != "recordType" {
-				t.Errorf("%T: %s", err, err)
-			}
-		}
+	err := addenda18.Validate()
+	// TODO: are we not expecting any errors here?
```

Comment:
return the fieldError

	if addenda18.recordType != "7" {
		fieldError("recordType", NewErrRecordType(7), addenda18.recordType)
	}